CoreScriptsRoot/Modules/DeveloperConsoleModule.lua

```diff
@@ -2929,4 +2952,61 @@ do
 	end
 end
 
-return DeveloperConsole
+--[[ Module Table ]]--
+-- We only create the dev console if we need it; user toggles visibility.
+
+local DevConsoleModuleTable = {}
+local myDeveloperConsole = nil
+
+-- Tenfoot Interface set up
+local function onDevConsoleVisibilityChanged(isVisible)
+	local blockMenuActionName = "blockMenuAction"
+	local closeDevConsoleActionName = "closeDevConsoleAction"
+	local selectionParentName = "devConsoleSelectionGroup"
+
+	local function closeDevConsole(actionName, inputState, inputObject)
+		if inputState == Enum.UserInputState.End then
+			myDeveloperConsole:SetVisible(false)
+		end
+	end
+
+	if isVisible then
+		-- block menu open input while dev console is open
+		ContextActionService:BindCoreAction(blockMenuActionName, function() end, false, Enum.KeyCode.ButtonStart)
+
+		local menuModule = require(Modules.Settings.SettingsHub)
+		menuModule:SetVisibility(false, true)
+		ContextActionService:BindCoreAction(closeDevConsoleActionName, closeDevConsole, false, Enum.KeyCode.ButtonB)
+
+		GuiService:AddSelectionParent(selectionParentName, myDeveloperConsole.Frame)
+		GuiService.SelectedCoreObject = myDeveloperConsole.CurrentOpenedTab
+	else
+		ContextActionService:UnbindCoreAction(closeDevConsoleActionName)
+		ContextActionService:UnbindCoreAction(blockMenuActionName)
+
+		GuiService:RemoveSelectionGroup(selectionParentName)
+		GuiService.SelectedCoreObject = nil
+	end
+end
+
+local function getDeveloperConsole()
+	if not myDeveloperConsole then
+		local permissions = DeveloperConsole.GetPermissions()
+		local messagesAndStats = DeveloperConsole.GetMessagesAndStats(permissions)
+
+		myDeveloperConsole = DeveloperConsole.new(RobloxGui, permissions, messagesAndStats)
+
+		if isTenFootInterface then
+			myDeveloperConsole.VisibleChanged:connect(onDevConsoleVisibilityChanged)
+		end
+	end
+
+	return myDeveloperConsole
+end
+
+function DevConsoleModuleTable:ToggleVisibility()
```

Comment:
It may be better in the long run to just expose the GetVisible and SetVisible functions. Since toggle only really works for this use case